### PR TITLE
Unit tests for encoding detection

### LINKED ISSUE #37190: StreamReader detectEncodingFromByteOrderMarks needs tests
We don't check that the detection algorithm works as expected. At the very least we should check that we set the encoding correctly based on the data stream. It would also be good to actually read data out in the different encodings.

StreamReader tests are in System.IO.

### PR DESCRIPTION
**Pull request by davidkaya**
Added unit tests for encoding detection

Closes #37190

### CODE REVIEW

File: src/System.IO/tests/StreamReader/StreamReader.CtorTests.cs [hunk 1]
```diff
@@ -42,10 +46,97 @@ public static void CreationFromMemoryStreamWithEncodingTrue()
             var ms2 = new MemoryStream();
             ms2.Write(new byte[] { 65, 66, 67, 68 }, 0, 4);
             ms2.Position = 0;
-            var sr2 = new StreamReader(ms2, false);
+            var sr2 = new StreamReader(ms2, true);
```
Comment (davidkaya) [inline src/System.IO/tests/StreamReader/StreamReader.CtorTests.cs]: I think this was incorrect before (there was a same test above and also based on the unit test name).

**Comment by davidkaya** [comment]
@JeremyKuhne The build failed multiple times on different hosts (e.g. at the Send to Helix step), were there some changes related to builds? Or am I just unlucky?

**Comment by JeremyKuhne** [comment]
> Or am I just unlucky?

Things appear to a bit flaky. Kicking again.

**Comment by danmosemsft** [comment]
Thansk @davidkaya , normally PR's don't get stuck this long. We're trying to do better.

**Pull request merged**